inst/tutorials/chapter3/chapter3.Rmd

```diff
@@ -145,32 +173,156 @@ y <- rlnorm(10000,meanlog=3.5,sdlog=1)
+
+shinyUI(fluidPage(
+  sidebarPanel(selectInput(inputId = "dist", # Dropdown Menu to choose a distribution
+                label = "Distribution:",
+                choices = c(Normal      = "rnorm",
+                            Lognormal   = "rlnorm",
+                            Uniform     = "runif",
+                            Beta        = "rbeta",
+                            Pareto      = "rparet",
+                            Arcsine     = "rarcsine",
+                            Ratio2Normals = "rationorm"),
+                selected = "rnorm"),
+               
+               # Change interactive parameters depending on the chosen distribution
+               
+               conditionalPanel(condition = "input.dist == 'runif'",
+                                sliderInput("range_unif", "Range", min = -100, max = 100, value = c(-5, 5))),
+               
+               conditionalPanel(condition = "input.dist == 'rbeta'",
+                                sliderInput("alpha", "Alpha", min = 0, max = 1, step = 0.01, value = .5),
+                                sliderInput("beta", "Beta", min = 0, max = 1, step = 0.01, value = .5)),
+               
+               conditionalPanel(condition = "input.dist == 'rnorm'",
+                                sliderInput("mean_norm", "Mean", min = -5, max = 5, step = .1, value = 0),
+                                sliderInput("std_norm", "Standard Deviation", min = 0.01, max = 5, step = 0.1, value = 1)),
+               
+               conditionalPanel(condition = "input.dist == 'rparet'",
+                                sliderInput("location", "Location", min = 0.1, max = 10, step = .1, value = 1),
+                                sliderInput("shape", "Shape", min = 0.1, max = 5, step = 0.1, value = 1)),
+               
+               conditionalPanel(condition = "input.dist == 'rarcsine'"),
+               
+               conditionalPanel(condition = "input.dist == 'rationorm'",
+                                sliderInput("a", "Shift Numerator", min = -5, max = 5, step = .1, value = -2),
+                                sliderInput("b", "Shift Denominator", min = -5, max = 5, step = .1, value = -3)),
+               
+               conditionalPanel(condition = "input.dist == 'rlnorm'",
+                                sliderInput("mean_lnorm", "Mean", min = -5, max = 5, step = .1, value = 0),
+                                sliderInput("std_lnorm", "Standard Deviation", min = 0.01, max = 5, step = 0.1, value = 1))),
+  
+  mainPanel(
+    plotOutput("dPlot") #plot
+  )))
+```
+
+```{r, context="server"}
+output$dPlot <- renderPlot({
+  
+  if (input$dist == 'rnorm'){ # Normal
+    range <- seq(-10, 10, .1)
+    plot(range,
+         dnorm(range, mean = input$mean_norm, sd = input$std_norm), "l", col = "navyblue", ylab = "Density", xlab = "y", main = "Normal Distribution")
+    abline(v=input$mean_norm ,col="red",lw=3)
+    abline(v=input$mean_norm ,col="blue",lw=1)
```

Comment:
i would use `lw=3` for median as well.

Reply:
That is what I went with at first. The problem here is that setting `lw=3` for both median and mean is confusing since we're only able to see one of the two when they overlap (for symmetric distributions)

I agree, there are better ways to go about this (like changing the legend dynamically?)